src/DynamoCoreWpf/Utilities/CrashReportTool.cs

```diff
@@ -0,0 +1,257 @@
+    {
+        public bool SendLogFile = true;
+        public bool SendSettingsFile = true;
+        public bool SendDynFile = true;
+        public bool SendRecordedCommands = true;
+
+        internal DynamoViewModel viewModel;
+
+        internal CrashReportArgs(DynamoViewModel dynamoViewModel)
+        {
+            viewModel = dynamoViewModel;
+        }
+    }
+
+    internal class CrashReportTool
+    {
+        static CrashReportTool()
+        {
+            try
+            {
+                CerToolDir = GetCERToolLocation();
+            }
+            catch
+            {}
+        }
+
+        private static List<string> ProductsWithCER => new List<string>() { "Revit", "Civil", "Robot Structural Analysis" };
+        private static readonly string CERExe = "senddmp.exe";
+
+
+        private static string CerToolDir;
+
+        private enum MINIDUMP_TYPE
+        {
+            MiniDumpNormal = 0x00000000,
+            MiniDumpWithDataSegs = 0x00000001,
+            MiniDumpWithFullMemory = 0x00000002,
+            MiniDumpWithHandleData = 0x00000004,
+            MiniDumpFilterMemory = 0x00000008,
+            MiniDumpScanMemory = 0x00000010,
+            MiniDumpWithUnloadedModules = 0x00000020,
+            MiniDumpWithIndirectlyReferencedMemory = 0x00000040,
+            MiniDumpFilterModulePaths = 0x00000080,
+            MiniDumpWithProcessThreadData = 0x00000100,
+            MiniDumpWithPrivateReadWriteMemory = 0x00000200,
+            MiniDumpWithoutOptionalData = 0x00000400,
+            MiniDumpWithFullMemoryInfo = 0x00000800,
+            MiniDumpWithThreadInfo = 0x00001000,
+            MiniDumpWithCodeSegs = 0x00002000,
+            MiniDumpWithoutAuxiliaryState = 0x00004000,
+            MiniDumpWithFullAuxiliaryState = 0x00008000,
+            MiniDumpWithPrivateWriteCopyMemory = 0x00010000,
+            MiniDumpIgnoreInaccessibleMemory = 0x00020000,
+            MiniDumpWithTokenInformation = 0x00040000,
+            MiniDumpWithModuleHeaders = 0x00080000,
+            MiniDumpFilterTriage = 0x00100000,
+            MiniDumpWithAvxXStateContext = 0x00200000,
+            MiniDumpWithIptTrace = 0x00400000,
+            MiniDumpScanInaccessiblePartialPages = 0x00800000,
+            MiniDumpFilterWriteCombinedMemory,
+            MiniDumpValidTypeFlags = 0x01ffffff
+        };
+
+        [DllImport("kernel32.dll")]
+        private static extern uint GetCurrentThreadId();
+
+        [DllImport("dbghelp.dll")]
+        private static extern bool MiniDumpWriteDump(
+            IntPtr hProcess,
+            UInt32 ProcessId,
+            SafeHandle hFile,
+            MINIDUMP_TYPE DumpType,
+            ref MINIDUMP_EXCEPTION_INFORMATION ExceptionParam,
+            IntPtr UserStreamParam,
+            IntPtr CallbackParam);
+
+
+        /// <summary>
+        /// Struct mapping to MINIDUMP_EXCEPTION_INFORMATION for Win32 API
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential, Pack = 4)]
```

Comment:
maybe you answered this already, was this pack size documented somewhere or found by trial and error?  it appears to make this struct smaller than if the default on an x64 system - I think there would be 4 bytes of padding between the `ThreadId`  and `ExceptionPointers` if the default was used?
I guess this is to make the padding the same between architectures?

Reply:
The main reason is this (from msdn docs):
https://docs.microsoft.com/en-us/cpp/dotnet/how-to-marshal-structures-using-pinvoke?view=msvc-170 
```
there is no relationship between the native and managed version of the two structures other than their data layout. Therefore, it is vital that the managed version contains fields that are the same size and in the same order as the native version. (There is no mechanism for ensuring that the managed and native versions of the structure are equivalent, so incompatibilities will not become apparent until run time. It is the programmer's responsibility to ensure that the two structures have the same data layout.)

Because the members of managed structures are sometimes rearranged for performance purposes, it is necessary to use the [StructLayoutAttribute](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.structlayoutattribute) attribute to indicate that the structure are laid out sequentially. It is also a good idea to explicitly set the structure packing setting to be the same as that used by the native structure. (Although by default, Visual C++ uses an 8-byte structure packing for both managed code.)
```
I could not find info on how the native structure is packed...but I did find an example of c# pInvoking into the minidump function...and it specified packing with 4 bytes and it seems to work.
Example from here : https://www.pinvoke.net/default.aspx/dbghelp/MiniDumpWriteDump.html

Reply:
The default pack size  for x64 is 16. This is normally not used by the Win32 apis if they want to keep binary backward compatibility for some reason.  They way to figure out what to use is to look in the header files for Win32.  So when I look at `minidumpapiset.h` I see in the beginning:

```
#include <pshpack4.h>
```

and in the end:

```
#include <poppack.h>
```

So 4 it is.

Reply:
Ugh..I searched for that header on my entire windows folder and program files.
Only now I realize that it was on the x86 program files...Damn
Thanks @sm6srw